<launch>

   <arg name="device_name" default="HD1"/>
   <param name="publish_rate" type="int" value="100" />
   
 
   <group ns="$(arg device_name)">
	<!-- omni node publishes joint states on omni1_joint_states -->
        <node name="$(arg device_name)" pkg="geomagic_control" type="omni" output="screen">
            <param name="device_name" type="str" value="$(arg device_name)"/>
            
        </node>
	
	
        
        <param name="$(arg device_name)_robot_description" command="cat $(find omni_description)/urdf/omni.urdf" />

	<!-- robot_state_publisher reads Geomagic_joint_states and publishes corresponding tfs -->
        <node name="$(arg device_name)_robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
	  <!--<remap from="joint_states" to="$(arg device_name)/joint_states" />-->
	  <remap from="robot_description" to="$(arg device_name)_robot_description" />
	</node>
	</group>
</launch>

